stock_available_unreserved/models/product.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2018 Camptocamp SA
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.addons import decimal_precision as dp
+
+UNIT = dp.get_precision('Product Unit of Measure')
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    qty_available_not_res = fields.Float(
+        string='Quantity On Hand Unreserved',
+        digits=UNIT,
+        compute='_compute_product_available_not_res',
+    )
+
+    qty_available_stock_text = fields.Char(
+        compute='_compute_product_available_not_res',
+        string='Unreserved stock quantity',
+    )
+
+    @api.multi
+    @api.depends('product_variant_ids.qty_available_not_res')
+    def _compute_product_available_not_res(self):
+        no_new = self.filtered(lambda x: not isinstance(x.id, models.NewId))
+        for tmpl in no_new:
+            tmpl.qty_available_not_res = sum(
+                tmpl.mapped('product_variant_ids.qty_available_not_res')
+            )
+            tmpl.qty_available_stock_text = "/".join(
+                tmpl.mapped('product_variant_ids.qty_available_stock_text')
+            )
+
+    @api.multi
+    def action_open_quants_unreserved(self):
+        products = self.mapped('product_variant_ids').ids
+        quants = self.env['stock.quant'].search(
+            [
+                ('product_id', 'in', products),
+            ]
+        )
+        quant_ids = quants.filtered(
+            lambda x: x.product_id.qty_available_not_res > 0
+        ).ids
+        result = self.env.ref('stock.product_open_quants').read()[0]
+        result['domain'] = "[('id', 'in', {})]".format(quant_ids)
```

Comment:
not a requirement, just FTR: this 2 keys can be python lists and dict, no need to convert to strings ;)